### ISSUE TITLE
Large files not save properly in storage

### ISSUE THREAD

**Issue by rosenberg-c** [appwrite/appwrite#1226]
## 🐛 Bug Report

From Appwrite project page/upload file and also via a react web app with sdk.storage.createFile(file, read, write); ( The documentation says I dont need to supply read or write. This is not true as I have to supply it or I get Appwrite Internal 500 error)

With default settings the filesize limit is 10mb, but files only around 5mb is succesfully uploaded.
I have changed the limit to 100mb and the result is the same.

Large files only get its beginning and end saved.(however opening a wav file downloaded after a succesfull upload only has the beginning)
For example flac/wav/m4a and other audio formats gets cut.
files around 5mb seems fine, but larger files fails.

No error message is given

## To Reproduce

1. Upload a large file
2. Download and check the content.

## Expected behavior

I expected the uploaded files to remain the same as the original files.
I believe the upload only uploads a part of the data to the storage.

## Actual Behavior
1. Upload a file
2. Its says success
3. Download and check the file.
3. The file is broken.

## Your Environment

ubuntu 20.04
Appwrite version 0.8.0

**Comment by eldadfux** [comment]
@rosenberg-c I think this is related to a bug fixed here: https://github.com/appwrite/appwrite/pull/1189 we have changed the output buffer to match the total allowed upload size, this is a temp fix until we'll properly handle output chunks for big files.

You can try the master branch to see if the fix is working for you.

**Comment by rosenberg-c** [comment]
Great to hear! 
Im not really sure how to do that, since im running of the docker version.. 
Any idea when a docker version get released with the fix?

**Comment by eldadfux** [comment]
Here are instructions on how to setup Appwrite from the master branch:
https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md#setup-from-source

We're still not sure when the next version will be release, I'll update here as soon as we have a date or a release.

**Comment by rosenberg-c** [comment]
> Here are instructions on how to setup Appwrite from the master branch:
> https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md#setup-from-source
> 
> We're still not sure when the next version will be release, I'll update here as soon as we have a date or a release.

Thank you for the suggestion, I tried it, but I got an error message while compiling:

```g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make: *** [Makefile:209: ext-src/php_swoole.lo] Error 1
ERROR: Service 'appwrite-worker-mails' failed to build: The command '/bin/sh -c git clone --depth 1 --branch $PHP_REDIS_VERSION https://github.com/phpredis/phpredis.git &&   cd phpredis &&   phpize &&   ./configure &&   make && make install &&   cd .. &&   git clone --depth 1 --branch $PHP_SWOOLE_VERSION https://github.com/swoole/swoole-src.git &&   cd swoole-src &&   phpize &&   ./configure --enable-http2 &&   make && make install &&   cd .. &&   git clone --depth 1 --branch $PHP_IMAGICK_VERSION https://github.com/Imagick/imagick &&   cd imagick &&   phpize &&   ./configure &&   make && make install &&   cd .. &&   git clone --depth 1 --branch $PHP_YAML_VERSION https://github.com/php/pecl-file_formats-yaml &&   cd pecl-file_formats-yaml &&   phpize &&   ./configure &&   make && make install &&   cd .. &&   git clone --depth 1 --branch $PHP_MAXMINDDB_VERSION https://github.com/maxmind/MaxMind-DB-Reader-php.git &&   cd MaxMind-DB-Reader-php &&   cd ext &&   phpize &&   ./configure &&   make && make install &&   cd ../..' returned a non-zero code: 2```

I suspect this is due my limited server running out of memory.. (1GHz , 1GB ram)

Im glad you could cut the memory requirement down for v 0.8.0 (and hoping you can maintan that limit), since I can keep the vm cost down until I really need more power.

**Comment by eldadfux** [comment]
This is now fixed and available as part of Appwrite 0.9:
https://dev.to/appwrite/announcing-appwrite-0-9-the-open-source-firebase-alternative-53ho 🥳 🎉
